spec/models/track_spec.rb

```diff
@@ -84,25 +84,85 @@
+    describe '#start_date_before_conference_end_date' do
+      before :each do
+        @conference = create(:conference, start_date: 1.day.ago, end_date: 2.days.from_now)
+      end
+
+      context 'is valid' do
+        it 'when the track\'s start date is before the conference\'s end date' do
+          track = build(:track, start_date: Date.today, end_date: Date.tomorrow, program: @conference.program)
+          expect(track.valid?).to eq true
         end
+      end
 
+      context 'is invalid' do
         it 'when the track\'s start date is after the conference\'s end date' do
           track = build(:track, start_date: 3.days.from_now, end_date: 4.days.from_now, program: @conference.program)
           expect(track.valid?).to eq false
           expect(track.errors[:start_date]).to eq ["can't be after the conference end date (#{2.days.from_now.to_date})"]
         end
+      end
+    end
+
+    describe '#end_date_after_conference_start_date' do
+      before :each do
+        @conference = create(:conference, start_date: 1.day.ago, end_date: 2.days.from_now)
+      end
+
+      context 'is valid' do
+        it 'when the track\'s end date is after the conference\'s start date' do
+          track = build(:track, start_date: Date.today, end_date: Date.tomorrow, program: @conference.program)
+          expect(track.valid?).to eq true
+        end
+      end
+
+      context 'is invalid' do
+        it 'when the track\'s end date is before the conference\'s start date' do
+          track = build(:track, start_date: 3.days.ago, end_date: 2.days.ago, program: @conference.program)
+          expect(track.valid?).to eq false
+          expect(track.errors[:end_date]).to eq ["can't be before the conference start date (#{1.day.ago.to_date})"]
+        end
+      end
+    end
+
+    describe '#end_date_before_conference_end_date' do
+      before :each do
+        @conference = create(:conference, start_date: 1.day.ago, end_date: 2.days.from_now)
+      end
+
+      context 'is valid' do
+        it 'when the track\'s start date is after the conference start date' do
+          track = build(:track, start_date: Date.today, end_date: Date.tomorrow, program: @conference.program)
+          expect(track.valid?).to eq true
+        end
+      end
 
+      context 'is invalid' do
         it 'when the track\'s end date is after the conference\'s end date' do
           track = build(:track, start_date: Date.today, end_date: 3.days.from_now, program: @conference.program)
           expect(track.valid?).to eq false
           expect(track.errors[:end_date]).to eq ["can't be after the conference end date (#{2.days.from_now.to_date})"]
         end
+      end
+    end
+
+    describe '#start_date_before_end_date' do
+      before :each do
+        @conference = create(:conference, start_date: 1.day.ago, end_date: 2.days.from_now)
+      end
 
+      context 'is valid' do
+        it 'when the track\'s start date is before its end date' do
```

Comment:
This can probably be combined with https://github.com/openSUSE/osem/pull/1639/files#diff-c2212054e3536a54e738fecea403adffR117

so that we are testing track dates within conference dates (I would even have it start on conf.start_date and end on conf.end_date which is our edge case).

And that would be enough for our tests, since we will test the start_date before end_date in the other validation.